keanu-project/src/main/java/io/improbable/keanu/vertices/dbl/nonprobabilistic/diff/LogProbGradientCalculator.java

```diff
@@ -0,0 +1,160 @@
+package io.improbable.keanu.vertices.dbl.nonprobabilistic.diff;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import com.google.common.base.Preconditions;
+
+import io.improbable.keanu.network.LambdaSection;
+import io.improbable.keanu.tensor.Tensor;
+import io.improbable.keanu.tensor.TensorShape;
+import io.improbable.keanu.tensor.dbl.DoubleTensor;
+import io.improbable.keanu.vertices.Probabilistic;
+import io.improbable.keanu.vertices.Vertex;
+import io.improbable.keanu.vertices.VertexId;
+import io.improbable.keanu.vertices.dbl.Differentiator;
+import io.improbable.keanu.vertices.dbl.DoubleVertex;
+
+public class LogProbGradientCalculator {
```

Comment:
Do we need to call this `Calculator`? I think it's pretty obvious it's going to calculate something! Can we just call it `LogProbGradient`?

Reply:
It's meant to be consistent with `ProbabilityCalculator`, which should probably be called `LogProbCalculator` instead. The calculator part is meant to make it sound like a object that does something with LogProb not an object that *is* the LogProb.

Reply:
Okay sounds good